immuni_analytics/helpers/safety_net.py

```diff
@@ -199,54 +259,10 @@ def verify_attestation(
     :param operational_info: the device operational information.
     :raises: SafetyNetVerificationError.
     """
-    try:
-        header = _get_jws_header(safety_net_attestation)
-    except (JSONDecodeError, binascii.Error, IndexError, ValueError) as exc:
-        _LOGGER.warning(
-            "Could not retrieve header from jws token.",
-            extra=dict(error=str(exc), safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-
-    if not (certificates_str := header.get("x5c", None)):  # pylint: disable=superfluous-parens
-        _LOGGER.warning(
-            "Could not retrieve certificates from jws header.",
-            extra=dict(safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-    try:
-        certificates = [base64.b64decode(c) for c in certificates_str]
-    except binascii.Error as exc:
-        _LOGGER.warning(
-            "Could not decode jws header certificates.",
-            extra=dict(error=str(exc), safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-
-    try:
-        leaf_certificate = _verify_certificates(certificates)
-    except certvalidator.errors.ValidationError as exc:
-        _LOGGER.warning(
-            "Could not validate the certificates chain.",
-            extra=dict(error=str(exc), safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-    try:
-        _verify_signature(safety_net_attestation, leaf_certificate)
-    except DecodeError as exc:
-        _LOGGER.warning(
-            "Could not verify jws signature.",
-            extra=dict(error=str(exc), safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-
-    try:
-        payload = _get_jws_payload(safety_net_attestation)
-    except (JSONDecodeError, binascii.Error, IndexError, ValueError) as exc:
-        _LOGGER.warning(
-            "Could not retrieve payload from jws token.",
-            extra=dict(error=str(exc), safety_net_attestation=safety_net_attestation),
-        )
-        raise SafetyNetVerificationError()
-
+    header = _get_jws_header(safety_net_attestation)
+    certificates = _get_certificates(header)
+    leaf_certificate = _get_leaf_certificate_x509(certificates[0])
```

Comment:
I would pass the whole array here. The leaf can be retrieved by the logic

Reply:
IMHO, it made clearer that this method only needs the first, and the `other_certificates_bytes` in the subsequent method would get the rest.

Yet, do you agree that if I'm changing this method to get the whole certificates array, we should do the same in the call below too?

Reply:
Done ✅